lib/puppetserver/ca/config/puppet.rb

```diff
@@ -46,20 +46,33 @@ def initialize(supplied_config_path = nil)
         # on Windows are unsupported.
         # Note that Puppet Server runs as the [pe-]puppet user but to
         # start/stop it you must be root.
-        def user_specific_conf_dir
-          @user_specific_conf_dir ||=
-            if Puppetserver::Ca::Utils::Config.running_as_root?
+        def running_as_root?
+          @as_root ||= Puppetserver::Ca::Utils::Config.running_as_root?
+        end
+
+        def user_specific_puppet_confdir
+          @user_specific_puppet_confdir ||=
+            if running_as_root?
               '/etc/puppetlabs/puppet'
             else
               "#{ENV['HOME']}/.puppetlabs/etc/puppet"
             end
         end
 
-        def user_specific_conf_file
-          user_specific_conf_dir + '/puppet.conf'
+        def user_specific_puppetconf_file
+          user_specific_puppet_confdir + '/puppet.conf'
+        end
+
+        def user_specific_puppetserver_confdir
+          @user_specific_puppetserver_confdir ||=
+            if running_as_root?
+              '/etc/puppetlabs/puppetserver'
+            else
+              "#{ENV['HOME']}/.puppetserver"
```

Comment:
This is weird, I have it locally, but I'm not sure a local dev environment ever actually uses it. ie, logging goes to stdout the `puppetserver.conf` is in the `./dev` directory of the repo and everything is under `~/.puppetlabs/{etc,opt}`